packages/mux-player/src/html.ts

```diff
@@ -108,24 +108,29 @@ export function processPart(part: Part, value: unknown): void {
     processPropertyIdentity(part, value);
 }
 
-const templates = new WeakMap<TemplateStringsArray, HTMLTemplateElement>();
+const templates = new Map<string, HTMLTemplateElement>();
 const renderedTemplates = new WeakMap<Node | ChildNodePart, HTMLTemplateElement>();
 const renderedTemplateInstances = new WeakMap<Node | ChildNodePart, TemplateInstance>();
 export class TemplateResult {
+  public readonly stringsKey: string;
+
   constructor(
     public readonly strings: TemplateStringsArray,
     public readonly values: unknown[],
     public readonly processor: any
-  ) {}
+  ) {
+    // Use a control character to join the expression boundaries.
+    this.stringsKey = this.strings.join('\x01');
```

Comment:
question(blocking): since `templates` is a module-scoped variable, what happens if two template strings in different parts of the codebase are identical? e.g. if this exists in two places:
```js
html`<style>${myStyles}</style>`
```

Reply:
it will use the same HTMLTemplateElement but most of the times it will be cloned when this condition returns false
https://github.com/muxinc/elements/blob/031a89a0d8a46d715b9ad778a732b720c7694ab5/packages/mux-player/src/html.ts#L140-L140

if it's true it's likely a condition like 

```js
return bool ? html`<style>${myStyles}</style>` : html`<style>${myPaint}</style>`
```

where only the expression value will update.